src/custom/components/Header/index.tsx

```diff
@@ -88,17 +76,74 @@ export default function Header() {
   const isUpToLarge = useMediaQuery(upToLarge)
 
   const [isMobileMenuOpen, setIsMobileMenuOpen] = useState(false)
-  const handleMobileMenuOnClick = () => setIsMobileMenuOpen(!isMobileMenuOpen)
+  const handleMobileMenuOnClick = useCallback(
+    () => isUpToLarge && setIsMobileMenuOpen(!isMobileMenuOpen),
+    [isUpToLarge, isMobileMenuOpen]
+  )
 
   // Toggle the 'noScroll' class on body, whenever the orders panel is open.
   // This removes the inner scrollbar on the page body, to prevent showing double scrollbars.
   useEffect(() => {
-    isOrdersPanelOpen ? document.body.classList.add('noScroll') : document.body.classList.remove('noScroll')
-    isUpToLarge && isMobileMenuOpen
-      ? document.body.classList.add('noScroll')
-      : document.body.classList.remove('noScroll')
+    isOrdersPanelOpen ? toggleBodyClass('noScroll', true) : toggleBodyClass('noScroll', false)
+    isUpToLarge && isMobileMenuOpen ? toggleBodyClass('noScroll', true) : toggleBodyClass('noScroll', false)
   }, [isOrdersPanelOpen, isMobileMenuOpen, isUpToLarge])
 
+  const getMainMenu = useMemo(
+    () =>
+      MAIN_MENU.map(({ title, url, externalURL, items }: MAIN_MENU_TYPE, index) =>
```

Comment:
I'm sure @anxolin will love this part 😅 

Anyway, would you be able to simplify this a bit?
Maybe use intermediate components or callbacks to reduce a bit the conditional chaining

Reply:
@alfetopito If we can simplify it further, that'd be great. Would any of you mind checking PR https://github.com/cowprotocol/cowswap/pull/596 (the current last one) and suggest how to further simplify? I know I did some changes up to the last PR.

Reply:
Yes, this is way too complex...

we can break it up a bit into smaller pieces. A map inside another map and with all these ternary operators are big red 🚩